feat: e-lab design changes

simon-hng opened the pull request.

Commit:
added timeline example - colors would have to be changed

Commit:
mobile changes

Commit:
Made the timeline to a component with subcomponent

simon-hng commented on components/Timeline.tsx:
```diff
@@ -0,0 +1,112 @@
+                    ({duration})
+                </h2>
+                <div className="relative flex-1 mb-10 bg-white border-b-4 border-blue-200 shadow dark:border-gray-700 rounded-3xl dark:bg-gray-900">
+                    <img
```
Should be next/image for image optimization

simon-hng commented on components/Timeline.tsx:
```diff
@@ -0,0 +1,112 @@
+                    title="Formation and Exploration"
+                    date="2nd October - 15th October 2023"
+                    duration="2 Weeks"
+                    imageSrc="/assets/e-lab/phases/co-founder.jpg"
```
Image src seems to be broken in the preview deployment

timonschramm replied:
Where do you see that it is broken, for me it looked fine, where could I check?

simon-hng replied:
https://website-nextjs-git-venture-timon-softdevtumai.vercel.app/e-lab

simon-hng commented on components/Timeline.tsx:
```diff
@@ -0,0 +1,112 @@
+            <div className="flex flex-col items-center w-10 mr-4 md:w-24">
+                <div>
+                    <div className="flex items-center justify-center w-8 h-8 bg-blue-200 rounded-full dark:bg-gray-600">
+                        <div className="w-4 h-4 bg-blue-600 rounded-full dark:bg-blue-400"></div>
```
We don't use dark/light theming yet

simon-hng commented on components/Timeline.tsx:
```diff
@@ -0,0 +1,112 @@
+                        <p className="mb-2 text-xl font-bold text-gray-600 dark:text-gray-400">
+                            {title}
+                        </p>
+                        <p className="text-gray-700 dark:text-gray-500">
```
Text-gray has insufficient contrast to background color

simon-hng reviewed (commented):
Please format with prettier

simon-hng commented:
You can add framer Motion to add scroll animations.

E.g. Timeline elements fade in on scroll  

https://www.framer.com/motion/scroll-animations/

Commit:
used prettier, framer, next/image, removed dark/white theme/ made gray stronger for better opacity

Commit:
Prettier fix

Commit:
changed the next/image and renamed folder

Commit:
made some mobile fixes, but framer seems to make some problems

Commit:
aligned the design, margins...

Commit:
changed the max-width and margins of the section component

Commit:
timeline is now aligned with other components on other pages than the E-Lab

simon-hng commented on components/Footer.tsx:
```diff
@@ -10,68 +11,72 @@ import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-            width="128"
-            height="32"
-          />
+      {/* <Section className="mx-auto max-w-7xl "> */}
```
dead code

simon-hng commented on components/Timeline.tsx:
```diff
@@ -0,0 +1,114 @@
+        <h2 className="mb-4 inline-block rounded-3xl bg-gradient-to-r from-blue-500 to-blue-900 px-4 py-2 text-xs font-medium text-gray-100">
+          {date} <br />({duration})
+        </h2>
+        <div className="relative mb-10 flex flex-1 flex-col rounded-3xl border-b-4 border-blue-200 bg-white shadow">
```
The border at the bottom looks a bit weird to me

simon-hng commented on components/Timeline.tsx:
```diff
@@ -0,0 +1,114 @@
+  );
+}
+
+export default function Timeline() {
```
We try to keep everything in components/ reusable. In this case we this component should have the Props

```ts
interface TimelineProps {
phases: {
  title: string,
  date: string,
  duration: string,
  image: string,
  text: string,
  }[]
}
```

The data that is used should be on the page

simon-hng committed:
feat: add spline header animation

simon-hng committed:
feat: add section content

simon-hng committed:
feat: improve mobile

simon-hng committed:
refactor: benefits component

simon-hng committed:
feat: add benefits and sponsors section

simon-hng committed:
feat: add timeline component

simon-hng committed:
fix(Benefit.tsx): increase grid gap to 16px for better spacing between benefits
fix(Header.tsx): change background color of overlay to bg-purple-950/80 for better contrast
fix(Header.tsx): change background color of header when scrolled and not navOpen to bg-purple-950/90 for better contrast
fix(Section.tsx): change padding to p-8 for consistent padding on all sides
fix(e-lab.tsx): add margin bottom of 8px to the first div in Section component for better spacing
fix(e-lab.tsx): change background color of Section component to bg-purple-950 for better contrast
fix(e-lab.tsx): change text color of Section component to text-white for better contrast
fix(e-lab.tsx): add margin bottom of 16px to the first div in Section component for better spacing on small screens
fix(e-lab.tsx): add margin bottom of 32px to the first div in Section component for better spacing on large screens
fix(e-lab.tsx): change text color of h2 element to text-white for better contrast
fix(e-lab.tsx): change text color of h3 element to text-yellow-500 for better contrast
fix(e-lab.tsx): remove width and height attributes from iframe element for responsive sizing
fix(e-lab.tsx): add aspect-video class to iframe element for responsive sizing
fix(e-lab.tsx): add margin top of 24px to the last div in Section component for better spacing
fix(e-lab.tsx): add margin left of 16px to h2 element in Section component for better alignment on medium screens
fix(e-lab.tsx): add margin left of 0 to h2 element in Section component for better alignment on large screens
fix(e-lab.tsx): change background color of second Section component to bg-purple-950 for better contrast
fix(e-lab.tsx): change text color of second Section component to text-white for better contrast
fix(e-lab.tsx): add margin left of 16px to h2 element in second Section component for better alignment on medium screens
fix(e-lab.tsx): add margin left of 0 to h2 element in second Section component for better alignment on large screens
fix(e-lab.tsx): change background color of scrollbar to bg-purple-500/10 for better visibility
fix(e-lab

simon-hng committed:
feat: add e-lab sponsors

simon-hng committed:
feat: update apply now and footer

Max-vS commented:
the navbar is too large on mobile devices and forces horizontal scroll into whitespace
